flaky test KeystoreScannerTest.testKeystoreRemoval()

Test failure `KeystoreScannerTest.testKeystoreRemoval()` after merging PR #5042

```
2020-07-20 23:28:46.020:INFO :oejs.Server:main: jetty-10.0.0-SNAPSHOT; built: unknown; git: ; jvm 11.0.5+10-LTS
2020-07-20 23:28:46.023:INFO :oejus.SslContextFactory:main: x509=X509@3148f668(jetty,h=[jetty.eclipse.org],w=[]) for Server@6e005dc9[provider=null,keyStore=file:///home/lachlan/webtide/src/jetty-10.0.x/tests/test-integration/target/tests/oejt.KeyStoreScannerTest/testKeystoreRemoval/keystore,trustStore=null]
2020-07-20 23:28:46.026:INFO :oejs.AbstractConnector:main: Started ServerConnector@573906eb{SSL, (ssl, http/1.1)}{0.0.0.0:38515}
2020-07-20 23:28:46.027:INFO :oejs.Server:main: Started Server@571c5681{STARTING}[10.0.0-SNAPSHOT,sto=0] @16371ms
2020-07-20 23:28:48.043:WARN :oejus.KeyStoreScanner:Scanner-6: Keystore Reload Failed: java.lang.IllegalStateException: /home/lachlan/webtide/src/jetty-10.0.x/tests/test-integration/target/tests/oejt.KeyStoreScannerTest/testKeystoreRemoval/keystore is not a valid keystore
2020-07-20 23:28:48.054:WARN :oeji.ManagedSelector:qtp541285440-90: Could not accept java.nio.channels.SocketChannel[closed]: java.lang.IllegalStateException: SslContextFactory reload failed
2020-07-20 23:28:50.060:WARN :oeji.ManagedSelector:qtp541285440-86: Could not accept java.nio.channels.SocketChannel[closed]: java.lang.IllegalStateException: SslContextFactory reload failed
2020-07-20 23:28:50.064:INFO :oejus.SslContextFactory:Scanner-6: x509=X509@739341f3(jetty-test-keystore,h=[],w=[]) for Server@6e005dc9[provider=null,keyStore=file:///home/lachlan/webtide/src/jetty-10.0.x/tests/test-integration/target/tests/oejt.KeyStoreScannerTest/testKeystoreRemoval/keystore,trustStore=null]
2020-07-20 23:28:50.067:INFO :oejs.Server:main: Stopped Server@571c5681{STOPPING}[10.0.0-SNAPSHOT,sto=0]
2020-07-20 23:28:50.068:INFO :oejs.AbstractConnector:main: Stopped ServerConnector@573906eb{SSL, (ssl, http/1.1)}{0.0.0.0:0}

javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake

	at java.base/sun.security.ssl.SSLSocketImpl.handleEOF(SSLSocketImpl.java:1321)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1160)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1063)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:402)
	at java.base/sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:567)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:185)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:168)
	at org.eclipse.jetty.test.KeyStoreScannerTest.getCertificateFromServer(KeyStoreScannerTest.java:264)
	at org.eclipse.jetty.test.KeyStoreScannerTest.testKeystoreRemoval(KeyStoreScannerTest.java:175)
        ...
Caused by: java.io.EOFException: SSL peer shut down incorrectly
	at java.base/sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:167)
	at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:108)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1152)
```

--- a/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/KeyStoreScanner.java
+++ b/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/KeyStoreScanner.java
@@ -104,6 +104,16 @@ public class KeyStoreScanner extends ContainerLifeCycle implements Scanner.Discr
             reload();
     }
 
+    @ManagedOperation(value = "Scan for changes in the SSL Keystore", impact = "ACTION")
+    public void scan()
+    {
+        if (LOG.isDebugEnabled())
+            LOG.debug("scanning");
+
+        _scanner.scan();
+        _scanner.scan();
+    }
+
     @ManagedOperation(value = "Reload the SSL Keystore", impact = "ACTION")
     public void reload()
     {

--- a/tests/test-integration/src/test/java/org/eclipse/jetty/test/KeyStoreScannerTest.java
+++ b/tests/test-integration/src/test/java/org/eclipse/jetty/test/KeyStoreScannerTest.java
@@ -24,7 +24,6 @@ import java.nio.file.Path;
 import java.security.SecureRandom;
 import java.security.cert.Certificate;
 import java.security.cert.X509Certificate;
-import java.time.Duration;
 import java.util.Calendar;
 import javax.net.ssl.HttpsURLConnection;
 import javax.net.ssl.KeyManager;
@@ -57,10 +56,10 @@ import static org.junit.jupiter.api.Assertions.assertThrows;
 @ExtendWith(WorkDirExtension.class)
 public class KeyStoreScannerTest
 {
-    private static final int scanInterval = 1;
     public WorkDir testdir;
     private Server server;
     private Path keystoreDir;
+    private KeyStoreScanner keystoreScanner;
 
     @BeforeEach
     public void before()
@@ -99,8 +98,8 @@ public class KeyStoreScannerTest
         server.addConnector(connector);
 
         // Configure Keystore Reload.
-        KeyStoreScanner keystoreScanner = new KeyStoreScanner(sslContextFactory);
-        keystoreScanner.setScanInterval(scanInterval);
+        keystoreScanner = new KeyStoreScanner(sslContextFactory);
+        keystoreScanner.setScanInterval(0);
         server.addBean(keystoreScanner);
 
         server.start();
@@ -123,7 +122,7 @@ public class KeyStoreScannerTest
 
         // Switch to use newKeystore which has a later expiry date.
         useKeystore("newKeystore");
-        Thread.sleep(Duration.ofSeconds(scanInterval * 2).toMillis());
+        keystoreScanner.scan();
 
         // The scanner should have detected the updated keystore, expiry should be renewed.
         X509Certificate cert2 = getCertificateFromServer();
@@ -143,11 +142,11 @@ public class KeyStoreScannerTest
         try (StacklessLogging ignored = new StacklessLogging(KeyStoreScanner.class))
         {
             useKeystore("badKeystore");
-            Thread.sleep(Duration.ofSeconds(scanInterval * 2).toMillis());
+            keystoreScanner.scan();
         }
 
         // The good keystore is removed, now the bad keystore now causes an exception.
-        assertThrows(Throwable.class, () -> getCertificateFromServer());
+        assertThrows(Throwable.class, this::getCertificateFromServer);
     }
 
     @Test
@@ -163,15 +162,15 @@ public class KeyStoreScannerTest
         try (StacklessLogging ignored = new StacklessLogging(KeyStoreScanner.class))
         {
             useKeystore(null);
-            Thread.sleep(Duration.ofSeconds(scanInterval * 2).toMillis());
+            keystoreScanner.scan();
         }
 
         // The good keystore is removed, having no keystore causes an exception.
-        assertThrows(Throwable.class, () -> getCertificateFromServer());
+        assertThrows(Throwable.class, this::getCertificateFromServer);
 
         // Switch to use keystore2 which has a later expiry date.
         useKeystore("newKeystore");
-        Thread.sleep(Duration.ofSeconds(scanInterval * 2).toMillis());
+        keystoreScanner.scan();
         X509Certificate cert2 = getCertificateFromServer();
         assertThat(getExpiryYear(cert2), is(2020));
     }
@@ -195,7 +194,7 @@ public class KeyStoreScannerTest
         // Change the symlink to point to the newKeystore file location which has a later expiry date.
         Files.delete(keystorePath);
         Files.createSymbolicLink(keystorePath, useKeystore("newKeystore"));
-        Thread.sleep(Duration.ofSeconds(scanInterval * 2).toMillis());
+        keystoreScanner.scan();
 
         // The scanner should have detected the updated keystore, expiry should be renewed.
         X509Certificate cert2 = getCertificateFromServer();
@@ -220,7 +219,7 @@ public class KeyStoreScannerTest
 
         // Change the target file of the symlink to the newKeystore which has a later expiry date.
         useKeystore("newKeystore");
-        Thread.sleep(Duration.ofSeconds(scanInterval * 2).toMillis());
+        keystoreScanner.scan();
 
         // The scanner should have detected the updated keystore, expiry should be renewed.
         X509Certificate cert2 = getCertificateFromServer();

